src/promptflow/promptflow/_sdk/_utils.py

```diff
@@ -1196,3 +1199,95 @@ def parse_otel_span_status_code(value: int) -> str:
         return "Ok"
     else:
         return "Error"
+
+
+def _generate_meta_from_file(working_dir, source_path, entry, meta_dict, exception_list):
+    with _change_working_dir(working_dir), inject_sys_path(working_dir):
```

Comment:
Why do we need to modify the working dir? Can downstream code use absolute paths?

Because this may lead to downstream code modifications requiring knowledge that the upstream has modified the working directory, and it is easy to encounter some problems without knowledge.

Reply:
Changing working dir is to help resolve relative imports like `from .. import xxx` since we have assumption that all flow's code will execute under flow.dag.yaml's directory. And this is a context manager, the working dir will be set back after flow meta generated.